sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -2427,4 +2427,14 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
       spark.read.option("mode", "PERMISSIVE").option("encoding", "UTF-8").json(Seq(badJson).toDS()),
       Row(badJson))
   }
+
+  test("SPARK-24190: restrictions for JSONOptions in read") {
+    val exception = intercept[IllegalArgumentException] {
+      spark.read
+        .option("encoding", "UTF-16")
+        .json(testFile("test-data/utf16LE.json"))
```

Comment:
super minor, but to more closely match the previous test maybe explicitly set multi-line to false explicitly.

Reply:
Actually the exception must be thrown for both cases - per-line and multiLine mode. I will check both mode, and both encoding `UTF-16` and `UTF-32`